stats/cloud/collector.go

```diff
@@ -250,21 +250,34 @@ func (c *Collector) Run(ctx context.Context) {
 	}
 }
 
-func useCloudTags(source *httpext.Trail) *httpext.Trail {
-	name, nameExist := source.Tags.Get("name")
-	url, urlExist := source.Tags.Get("url")
+// replaceTags overwrites the url tag to match the value of the name tag.
+// See #1220.
+func replaceTags(tags *stats.SampleTags) *stats.SampleTags {
+	name, nameExist := tags.Get("name")
+	url, urlExist := tags.Get("url")
 	if !nameExist || !urlExist || name == url {
-		return source
+		return tags
 	}
 
-	newTags := source.Tags.CloneTags()
+	newTags := tags.CloneTags()
 	newTags["url"] = name
 
+	return stats.IntoSampleTags(&newTags)
+}
+
+func replaceTagsHTTP(source *httpext.Trail) *httpext.Trail {
 	dest := new(httpext.Trail)
 	*dest = *source
-	dest.Tags = stats.IntoSampleTags(&newTags)
+	dest.Tags = replaceTags(source.GetTags())
 	dest.Samples = nil
+	return dest
+}
 
+func replaceTagsNet(source *netext.NetTrail) *netext.NetTrail {
+	dest := new(netext.NetTrail)
+	*dest = *source
+	dest.Tags = replaceTags(source.GetTags())
+	dest.Samples = nil
 	return dest
 }
 
```

Comment:
This way you will always make a copy of of the *`Trail`s. 
I propose you either make another function to check if it's needed at all(which is the first lines of `replaceTags`) or (IMO better) first to call `replaceTags` and if that returns a different `Tags` to then copy the *`Trail` and set the `Tags` and drop the `Samples`.

Reply:
Ah, indeed, thanks for the quick review. Will fix.

Reply:
While this is fixed in 484cafd2, the overall repetition and the additional check now is bothering me. I looked into ways of making it more generic, but given Go's restrictive type system (mostly a good thing) I didn't find an elegant way of doing that without adding more complexity (more interfaces or using `reflect`). But let me know if it can be improved.

Also, should we replace the tags on the `default` case as well?